src/service_daemon.rs

```diff
@@ -117,6 +117,8 @@ pub type Metrics = HashMap<String, i64>;
 pub struct ServiceDaemon {
     /// Sender handle of the channel to the daemon.
     sender: Sender<Command>,
+    /// name of the network interface to use (use all interfaces if None)
+    interface_filter: Option<String>,
```

Comment:
Currently we store such state only in `Zeroconf`, it will be better to keep it that way, to be consistent. 

I see you also added this state in `Zeroconf`, so I think you don't need to add it here. For the only usage I can see is in `register()` method,  you can move the `insert_ipv4addr` work into the command receiver side, i.e. in `fn register_service`.